src/Jupyter/Extensions.cs

```diff
@@ -202,6 +202,18 @@ public static bool TryDecodeParameter<T>(this Dictionary<string, string> paramet
                 parameterValue = parameterValue.Substring(1, parameterValue.Length - 2);    // Strip off the enclosing quotes
                 parameterValue = parameterValue.Replace("\\\"", "\"");                      // Don't escape the interior quotes
             }
+
+            // This is a temporary fix in order to send JSON encoded data as strings to the Azure submission client
+            var intermediate = JsonConvert.DeserializeObject<ImmutableDictionary<string, object>>(parameterValue);
+            if (intermediate == null) {
+                return defaultValue;
+            } else {
+                var transformedIntermediate = new Dictionary<string, string>();
+                foreach (var (key, value) in intermediate) {
```

Comment:
I'd suggest using LINQ's ToDictionary here instead of accumulating manually.

Reply:
@cgranade it looks like `ToDictionary` takes an `Enumerable<T>` and turns it into a dictionary, which won't work here as `intermediate` is already a dictionary. Unless there's another LINQ method I'm overlooking?

Reply:
Dictionaries implement `IEnumerable` over key-value pairs, such that the following should work:
```csharp
var transformed = intermediate
    .ToDictionary(pair => pair.Key, pair => pair.Value.ToString())
    .ToImmutableDictionary();
```
You can even use `ToImmutableDictionary` directly, as it has an overload that takes key and value projectors (https://learn.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutabledictionary.toimmutabledictionary?view=net-6.0#system-collections-immutable-immutabledictionary-toimmutabledictionary-3(system-collections-generic-ienumerable((-0))-system-func((-0-1))-system-func((-0-2)))):
```csharp
var transformed = intermediate
    .ToImmutableDictionary(pair => pair.Key, pair => pair.Value.ToString());
```
(Please forgive typos that may prevent above from running as written; didn't test these suggestions yet.)

Reply:
Aha I see, thank you for clarifying! I'll get that change up momentarily then.